### PR TITLE
Hip 991 patch

### PR DESCRIPTION
**Pull request by netopyr**
**Description**:

Merged all changes from:

- https://github.com/hashgraph/hedera-improvement-proposal/pull/1041
- https://github.com/hashgraph/hedera-improvement-proposal/pull/1042

Added modifications from discussions

### CODE REVIEW

**Review by ty-swirldslabs — APPROVED** [review]
lgtm

**Review by mgarbs — APPROVED** [review]

File: HIP/hip-991.md [hunk 1]
```diff
@@ -167,16 +183,21 @@ message ConsensusCreateTopicTransactionBody {
...
     */
-  repeated ConsensusCustomFee custom_fees = 9;
+  repeated ConsensusCustomFee custom_fees = 10;
 }
 ```
 
 #### ConsensusUpdateTopicTransactionBody
 
-The `ConsensusUpdateTopicTransactionBody` message is updated to include the optional Fee Schedule Key and `custom_fees` property for specifying fixed fees during topic creation.
+The `ConsensusUpdateTopicTransactionBody` message is updated to include the optional Fee Schedule Key, the optional Fee Exempt Key List, and the `custom_fees` property for specifying fixed fees during topic creation.
```
Comment (kimbor) [inline HIP/hip-991.md:200]:
```suggestion
The `ConsensusUpdateTopicTransactionBody` message is updated to include the optional Fee Schedule Key, the optional Fee Exempt Key List, the `custom_fees` property for specifying fixed fees during topic creation, and the `key_verification_mode` property for specifying whether invalid keys are allowed.
```

File: HIP/hip-991.md [hunk 2]
```diff
@@ -105,24 +105,40 @@ We propose adding a fixed fee mechanism to the Hedera Consensus Service (HCS) fo
...
+* FEKL can be updated with a `ConsensusUpdateTopicTransaction` signed by the topic's Admin Key
+* FEKL has a maximum of `MAX_ENTRIES_FOR_FEE_EXEMPT_KEY_LIST` entries.
+* If a TopicMessageSubmitTransaction is submitted to the network and it contains a signature from a key included in the FEKL, no fees will be charged for that message.
+  
+* **Handling Duplicates**:
+   - If the FEKL list contains duplicate keys, the transaction will fail with an `FEKL_CONTAINS_DUPLICATED_KEYS` error. This ensures that duplicate entries are not silently ignored, preventing potential bugs or issues in the calling code.
 
-#### Fee Exclusions (SDK)
+* **Signatures for Invalid/Inactive/Deleted Accounts**:
+   - The FEKL list will only require keys to be formally valid as per protobuf specifications, meaning they must be correctly formatted key structures. These are keys, not accounts. Even if an account associated with a key is inactive, deleted, or non-existent, the key itself can still be added to the FEKL.
```
Comment (kimbor) [inline HIP/hip-991.md:128]: Is key_verification_mode meant to change this behavior?

File: HIP/hip-991.md (hunk 1 above)
Comment (kimbor) [reply HIP/hip-991.md:200]: key_verification_mode should be explained in the text above

File: HIP/hip-991.md [hunk 3]
```diff
@@ -105,24 +105,40 @@ We propose adding a fixed fee mechanism to the Hedera Consensus Service (HCS) fo
 * If the `allowCustomFeesPayment` flag is not set or is not true, the SDK returns an error.
 * The `allowCustomFeesPayment` flag exists only in the SDK and isn't enforced by the network.
 
+* **Allowance Transactions**: 
+   - An account submits a transaction to grant an allowance (total fees and maximum fee per message) to a topic for both HBAR and HTS tokens. This transaction type reuses the existing allowance concept, with no need for a separate delete allowance transaction. To remove an allowance, the user will submit the same transaction with the allowance values set to 0 or a dedicated flag indicating allowance removal.
```
Comment (kimbor) [inline HIP/hip-991.md:109]:
```suggestion
   - An account submits a transaction to grant an allowance (total fees and maximum fee per message) to a topic for both HBAR and HTS tokens. This transaction type reuses the existing allowance concept, with no need for a separate delete allowance transaction. To remove an allowance, the user will submit the same transaction with the allowance values set to 0.
```

Comment (kimbor) [reply HIP/hip-991.md:109]: I don't see a flag specified in the FeeScheduleAllowance messages below, so I'm thinking that we decided against a dedicated flag

**Pull request merged**